File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.java
class Solution {
    public int minDeletionSize(String[] strs) {
       int numColumns = 0;
        for (int i = 0; i < strs[0].length(); i++) { 
            int j = 0;
            while (j < strs.length - 1 && strs[j].charAt(i) <= strs[j + 1].charAt(i)) {
                j++;
            }
            if (j != strs.length - 1) {
                numColumns++;
            }
        }
        return numColumns;
    }
}

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        a=0
        for i in range(len(strs[0])): 
            ans=[]
            for j in range(len(strs)):
                ans.append(strs[j][i])
            if ans!=sorted(ans):
                a+=1
        return a
        